mne_connectivity/spectral/time.py

```diff
@@ -392,12 +420,40 @@ def spectral_connectivity_time(
         if hasattr(data, "annotations") and not annots_in_metadata:
             data.add_annotations_to_metadata(overwrite=True)
         metadata = data.metadata
-        # XXX: remove logic once support for mne<1.6 is dropped
-        kwargs = dict()
-        if "copy" in inspect.getfullargspec(data.get_data).kwonlyargs:
-            kwargs["copy"] = False
-        data = data.get_data(**kwargs)
-        n_epochs, n_signals, n_times = data.shape
+        if isinstance(data, BaseEpochs):
+            # XXX: remove logic once support for mne<1.6 is dropped
+            kwargs = dict()
+            if "copy" in inspect.getfullargspec(data.get_data).kwonlyargs:
+                kwargs["copy"] = False
+            data = data.get_data(**kwargs)
+            n_epochs, n_signals, n_times = data.shape
+        else:
+            freqs = data.freqs  # use freqs from EpochsTFR object
+            # Read mode from the EpochsTFR object
+            mode = "cwt_morlet" if data.method == "morlet" else data.method
+            # TFR objs will drop bad channels, so specify picking all channels
+            data = data.get_data(picks=np.arange(data.info["nchan"]))
+            if not np.iscomplexobj(data):
+                raise TypeError(
+                    "if `data` is an EpochsTFR object, it must contain complex-valued "
+                    "Fourier coefficients, such as that returned from "
+                    "Epochs.compute_tfr() with `output='complex'`"
+                )
+            n_epochs, n_signals = data.shape[:2]
+            n_times = data.shape[-1]
+            # Warn if mt_bandwidth & n_cycles default-valued (can't be read from TFR)
+            if mode == "multitaper":  # doesn't matter for cwt_morlet
+                if mt_bandwidth is None:
+                    warn(
+                        "`mt_bandwidth` is not specified; assuming 4.0 Hz was used to "
+                        "compute the TFR"
+                    )
+                if n_cycles == 7.0:
+                    warn(
+                        "`n_cycles` is the default value; assuming 7.0 was used to "
+                        "compute the TFR"
+                    )
```

Comment:
Maybe we should add the `mt_bandwidth` and `n_cycles` attributes to the `*TFR` objects?

Reply:
It would be nice to have. I'm happy to open a PR if people think it's justified.

Reply:
Provenance is usually nice to track and if it's easy enough to add to the class then we might as well. Might be worth a quick MNE issue as we might want them as attributes like `tfr.n_cycles` but maybe better would be something like a dict of params like `tfr.estimation_parameters = {'mt_bandwidth': 4, 'n_cycles': 5}` so that the `tfr` namespace itself doesn't get too busy.

Reply:
Sure, will open an issue first then.

Reply:
maybe a naive question, but if what you need is the taper weights, why not store those directly rather than storing the other estimation params necessary to recompute them?  It's not a perfect solution because TFRs computed from earlier versions of MNE and then saved to disk may still lack the necessary information... but the same is true of storing the estimation parameters as attributes (instead of the weights).

Reply:
Copying from the discussion in https://github.com/mne-tools/mne-python/issues/12851#issuecomment-2360912513 for reference:

> drammock: Since we already store weights for spectrum objects, I'm inclined to store weights (not estimation params) for TFRs too. That saves having to recompute them when users pass MNE objects (not arrays) to the connectivity functions, and is more internally consistent.

>> tsbinns: Sure, I see your reasoning, especially to keep it consistent with *Spectrum objects. I'll open a PR for this.